packages/core/src/utils/ComponentError.ts

```diff
@@ -0,0 +1,54 @@
+import {NativeModules} from "react-native";
+
+interface ComponentError extends Error {
+  componentStack: string;
+}
+
+const isJSXError = (error: Error): error is ComponentError => {
+  return "componentStack" in error;
+};
+
+/**
+ * Logs errors with additional React component stack information if available.
+ *
+ * This function checks if the provided error is a React Native rendering error
+ * (i.e., it includes a `componentStack` property). If the `componentStack` is
+ * present and non-empty, the error details, including the message and the component
+ * stack trace, are logged using `EmbraceManager.logHandledError`.
+ *
+ * **Important Notes:**
+ * - The `componentStack` provides a trace of the React component tree at the time
+ *   of the error but does not include line or column numbers.
+ * - This function is complementary to other logging mechanisms that capture more
+ *   detailed stack traces, including line and column numbers.
+ * - By default, in release builds, the screen names in the `componentStack` appear as
+ *   `Unknown`. To retain meaningful screen names in release builds, you must configure
+ *   Metro to preserve the names of React components during minification. For detailed
+ *   instructions, refer to https://docs.embrace.io/docs/react-native/features/tracking-error-component/.
+ *
+ * Example `componentStack` trace:
+ * ```
+ * at undefined (in App)
```

Comment:
same question from the docs PR, will these always be "at undefined"? If so we may want to remove them so that we log a cleaner output

Reply:
We chan change it to: "at anonymous"

Reply:
I'd like to understand a bit more where that is coming from, is it always going to be "at undefined" or are there cases where it will show different information?